Picked up from the distro-pkg-dev mailing list.
Patch done by Thomas Fitzsimmons (fitzsim@redhat.com)
http://article.gmane.org/gmane.comp.java.openjdk.distro-packaging.devel/78

--- openjdk/hotspot/build/linux/makefiles/gcc.make.orig	2007-06-17 16:46:45.000000000 -0400
+++ openjdk/hotspot/build/linux/makefiles/gcc.make	2007-06-17 16:47:12.000000000 -0400
@@ -46,18 +46,12 @@
 # Compiler flags
 
 # position-independent code
-PICFLAG = -fPIC
+PICFLAG = -fpic
 
 VM_PICFLAG/LIBJVM = $(PICFLAG)
 VM_PICFLAG/AOUT   =
 
-ifneq ($(BUILDARCH), i486)
 VM_PICFLAG        = $(VM_PICFLAG/$(LINK_INTO))
-else
-# PIC has significant overhead on x86, build nonpic VM for now.
-# Link JVM at a "good" base location to avoid unnecessary .text patching.
-JVM_BASE_ADDR     = 0x06000000
-endif
 
 CFLAGS += $(VM_PICFLAG)
 CFLAGS += -fno-rtti
